client_test/virtualfund_test.go

```diff
@@ -47,11 +47,66 @@ func TestVirtualFundIntegration(t *testing.T) {
 	chainServiceA := chainservice.NewMockChainService(chain, alice.Address())
 	chainServiceB := chainservice.NewMockChainService(chain, bob.Address())
 	chainServiceI := chainservice.NewMockChainService(chain, irene.Address())
+	chainServiceBr := chainservice.NewMockChainService(chain, brian.Address())
 	broker := messageservice.NewBroker()
 
 	clientA, _ := setupClient(alice.PrivateKey, chainServiceA, broker, logDestination, 0)
+	irene, _ := setupClient(irene.PrivateKey, chainServiceI, broker, logDestination, 0)
+	ivan, _ := setupClient(brian.PrivateKey, chainServiceBr, broker, logDestination, 0)
 	clientB, _ := setupClient(bob.PrivateKey, chainServiceB, broker, logDestination, 0)
-	clientI, _ := setupClient(irene.PrivateKey, chainServiceI, broker, logDestination, 0)
 
-	openVirtualChannels(t, clientA, clientB, clientI, 1)
+	openN_HopVirtualChannels(t, []client.Client{clientA, irene, ivan, clientB}, 1)
+}
+
+// openN_HopVirtualChannels connects the n given participants in a line of ledger channels,
+// then uses these ledger connections to open channels between the first participant
+// and each other participant.
+//
+// This makes channels with 0, 1, ..., [len(participants) - 1] hops.
+func openN_HopVirtualChannels(t *testing.T, participants []client.Client, channelsPerCounterparty uint) {
```

Comment:
Does this function have canonical casing? Seems jarring to have an underscore in there when it is otherwise camel cased?

Reply:
We hyphenate n-hop in plaintext, but the hyphen isn't an allowed character in an identifier.

Also, the single-letter N wanted capitalization, and the word hop did as well. Putting both together loses the effect of capitals separating words.

In short, I tried a few different casing / spacing things, but all felt bad.

Reply:
OK. I don't feel strongly. But I think `openNHopVirtualChannels` is fine.